src/test/java/uk/gov/crowncommercial/dts/scale/service/agreements/controller/AgreementControllerTest.java

```diff
@@ -79,6 +93,35 @@ public void testGetAgreementUnexpectedError() throws Exception {
         .andExpect(content().string(containsString("An error occurred processing the request")));
   }
 
+  @Test
+  public void testGetAgreementLotsSuccess() throws Exception {
+    LotDetail lot = new LotDetail();
+    lot.setNumber(LOT_NUMBER);
+    Set<Lot> mockLots = new HashSet<>(Arrays.asList(mockLot));
+    when(service.findAgreementByNumber(AGREEMENT_NUMBER)).thenReturn(mockCommercialAgreement);
+    when(mockCommercialAgreement.getLots()).thenReturn(mockLots);
+    when(converter.convertLotsToDTOs(mockLots)).thenReturn(Arrays.asList(lot));
+    this.mockMvc.perform(get("/agreements/" + AGREEMENT_NUMBER + "/lots"))
+        .andExpect(status().isOk()).andExpect(content().string(containsString(LOT_NUMBER)));
+  }
+
+  @Test
+  public void testGetAgreementLotsNotFound() throws Exception {
```

Comment:
Think this should be testing path `/agreements/{ca-number}/lots/{lot-number}` and expect error "Lot number '%s' for agreement number '%s' not found" ?
Could also upgrade a few of these to verify the structure and values of the JSON returned - e.g.
```
.andExpect(jsonPath("$.errors[0].status", is("404 NOT_FOUND")))
.andExpect(jsonPath("$.errors[0].title", is("Resource not found")))
.andExpect(jsonPath("$.errors[0].detail", is("Lot number '1' for agreement number 'RM1000' not found")));
```
etc..

Reply:
This is meant to test `/agreements/{ca-number}/lots` (rather than a specific lot, which is tested elsewhere) - but it is still wrong 😄 

I will update and update the assertion tests across the board to use jsonPath - much better 👍

Reply:
Ha.. maybe we should both get our eyesight tested sooner rather than later.. 👓